### PR TITLE
added the jenkinsfile for haimpoc

### PR DESCRIPTION
**Pull request by athiruma**
Haim POC: 

1. Cleanup:  **dry_run='yes'**
ec2_idle - alerts user
ec2_stop - alerts user
ebs_unattached - alerts user
Zombie_cluster_resource - no alert
Zombie_elastic_ips  - no alert
Zombie_nat_gateways  - no alert
S3_inactive  - no alert
Zombie_snapshots  - no alert
Empty_roles - no alert

2.  Tagging
Adding auto the following tags for every resource:
User
Manager [LDAP]
Budget [Constant]
Email [LDAP]
Project [Spreadsheet]
Environment  [Spreadsheet]

### CODE REVIEW

File: jenkins/poc/haim/run_policies.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+LOGS = os.environ.get('LOGS', 'logs')
+
+mandatory_tags_appeng = {'Budget': 'APPENG'}
+
+os.system(f"""echo "Running the tag_iam_user" """)
+os.system(f"""podman run --rm --name cloud-governance-poc-haim --net="host" -e account="APPENG" -e policy="tag_iam_user" -e AWS_ACCESS_KEY_ID="{AWS_ACCESS_KEY_ID_APPENG}" -e AWS_SECRET_ACCESS_KEY="{AWS_SECRET_ACCESS_KEY_APPENG}" -e user_tag_operation="update" -e SPREADSHEET_ID="{SPREADSHEET_ID}" -e GOOGLE_APPLICATION_CREDENTIALS="{GOOGLE_APPLICATION_CREDENTIALS}" -v "{GOOGLE_APPLICATION_CREDENTIALS}":"{GOOGLE_APPLICATION_CREDENTIALS}" -e LDAP_HOST_NAME="{LDAP_HOST_NAME}"  -e log_level="INFO" quay.io/ebattat/cloud-governance:latest""")
+
+
+regions = ['us-east-1', 'us-east-2', 'us-west-1', 'us-west-2', 'eu-central-1', 'ap-south-1', 'eu-north-1', 'ap-northeast-1', 'ap-southeast-1', 'ap-southeast-2', 'eu-west-3', 'sa-east-1']
```
[the remark this answers is not in the document]
Comment (athiruma) [reply jenkins/poc/haim/run_policies.py]: yes, ALL_ACTIVE_REGIONS=True it provide this ability

File: jenkins/poc/haim/run_policies.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+LOGS = os.environ.get('LOGS', 'logs')
+
+mandatory_tags_appeng = {'Budget': 'APPENG'}
+
+os.system(f"""echo "Running the tag_iam_user" """)
+os.system(f"""podman run --rm --name cloud-governance-poc-haim --net="host" -e account="APPENG" -e policy="tag_iam_user" -e AWS_ACCESS_KEY_ID="{AWS_ACCESS_KEY_ID_APPENG}" -e AWS_SECRET_ACCESS_KEY="{AWS_SECRET_ACCESS_KEY_APPENG}" -e user_tag_operation="update" -e SPREADSHEET_ID="{SPREADSHEET_ID}" -e GOOGLE_APPLICATION_CREDENTIALS="{GOOGLE_APPLICATION_CREDENTIALS}" -v "{GOOGLE_APPLICATION_CREDENTIALS}":"{GOOGLE_APPLICATION_CREDENTIALS}" -e LDAP_HOST_NAME="{LDAP_HOST_NAME}"  -e log_level="INFO" quay.io/ebattat/cloud-governance:latest""")
+
+
+os.system(f"""echo "Running the tag_resources" """)
```
Comment (ebattat) [inline jenkins/poc/haim/run_policies.py:20]: Is that including only tag resource ?
I ask because we need to run the policies in dry run = "yes" ?

Comment (athiruma) [reply jenkins/poc/haim/run_policies.py:20]: yes, it tags the resources.

File: cloud_governance/main/environment_variables.py [hunk 3]
```diff
@@ -165,6 +165,8 @@ def __init__(self):
         self._environment_variables_dict['EMAIL_ALERT'] = EnvironmentVariables.get_boolean_from_environment('EMAIL_ALERT', True)
         self._environment_variables_dict['MANAGER_EMAIL_ALERT'] = EnvironmentVariables.get_boolean_from_environment('MANAGER_EMAIL_ALERT', True)
 
+        self._environment_variables_dict['ALL_ACTIVE_REGIONS'] = EnvironmentVariables.get_boolean_from_environment('ALL_ACTIVE_REGIONS', False)
+
     @staticmethod
```
Comment (ebattat) [inline cloud_governance/main/environment_variables.py:170]: Did we remove it?

File: cloud_governance/main/environment_variables.py [hunk 4]
```diff
@@ -34,7 +34,7 @@ def __init__(self):
         # dynamic parameters - configure for local run
         # parameters for running policies
         self._environment_variables_dict['account'] = EnvironmentVariables.get_env('account', '').upper()
-        self._environment_variables_dict['AWS_DEFAULT_REGION'] = EnvironmentVariables.get_env('AWS_DEFAULT_REGION', '')
+        self._environment_variables_dict['AWS_DEFAULT_REGION'] = EnvironmentVariables.get_env('AWS_DEFAULT_REGION', 'us-east-2')
```
Comment (ebattat) [inline cloud_governance/main/environment_variables.py:37]: Why we need default?

File: jenkins/poc/haim/hourly/run_policies.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+
+os.system(f"""echo "Running the tag_resources" """)
+regions = ['us-east-1', 'us-east-2', 'us-west-1', 'us-west-2', 'eu-central-1', 'ap-south-1', 'eu-north-1', 'ap-northeast-1', 'ap-southeast-1', 'ap-southeast-2', 'eu-west-3', 'sa-east-1']
+for region in regions:
+    os.system(f"""podman run --rm --name cloud-governance-poc-haim -e account="APPENG" -e policy="tag_resources" -e AWS_ACCESS_KEY_ID="{AWS_ACCESS_KEY_ID_APPENG}" -e AWS_SECRET_ACCESS_KEY="{AWS_SECRET_ACCESS_KEY_APPENG}" -e AWS_DEFAULT_REGION="{region}" -e tag_operation="update" -e mandatory_tags="{mandatory_tags_appeng}" -e log_level="INFO" -v "/etc/localtime":"/etc/localtime" quay.io/ebattat/cloud-governance:latest""")
```
Comment (ebattat) [inline jenkins/poc/haim/hourly/run_policies.py:23]: Pls also add the policies

File: cloud_governance/main/environment_variables.py (hunk 4 above)
Comment (athiruma) [reply cloud_governance/main/environment_variables.py]: removed

File: cloud_governance/main/environment_variables.py (hunk 3 above)
Comment (athiruma) [reply cloud_governance/main/environment_variables.py]: removed

File: jenkins/poc/haim/daily/run_policies.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+
+regions = ['us-east-1', 'us-east-2', 'us-west-1', 'us-west-2', 'eu-central-1', 'ap-south-1', 'eu-north-1', 'ap-northeast-1', 'ap-southeast-1', 'ap-southeast-2', 'eu-west-3', 'sa-east-1']
+policies = get_policies()
+policies.sort()
+policies.remove('cost_explorer')
+policies.remove('cost_over_usage')
+policies.remove('monthly_report')
+policies.remove('cost_billing_reports')
+policies.remove('cost_explorer_payer_billings')
```
Comment (ebattat) [inline jenkins/poc/haim/daily/run_policies.py:38]: I think we should write it like this:

```
not_actions_policies = ['cost_explorer', 'cost_over_usage', 'monthly_report', 'cost_billing_reports', 'cost_explorer_payer_billings']


run_policies = list(set(policies) - set(not_actions_policies))
```

WDYT ?

**Review by ebattat — APPROVED** [review]
/LGTM

**Pull request merged**